parlai/agents/bert_ranker/bi_encoder_ranker.py

```diff
@@ -46,7 +46,7 @@ def __init__(self, opt, shared=None):
         super().__init__(opt, shared)
         # it's easier for now to use DataParallel when
         self.data_parallel = opt.get('data_parallel') and self.use_cuda
-        if self.data_parallel:
+        if self.data_parallel and shared is None:
```

Comment:
this is good but i'm just noticing that the `self.rank_loss` below is not correct. we should override `build_criterion` instead.

Reply:
Created a new issue: #2019.